src/Microsoft.Dnx.Runtime/Compilation/ICompilationSession.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Dnx.Runtime.Compilation
+{
+    public interface ICompilationSession : IDisposable
+    {
+        event Action<string> OnInputFileChanged;
+
+        Assembly CompileAndLoadProject(Project project, CompilationTarget target, IAssemblyLoadContext loadContext);
```

Comment:
Name? Yeah, I thought about just `LoadProject` but I wanted to make it clear what it does... maybe that's overkill though